src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/Analyzers/ConvertToLibraryImportAnalyzer.cs

```diff
@@ -139,6 +160,26 @@ private static void AnalyzeSymbol(SymbolAnalysisContext context, INamedTypeSymbo
             context.ReportDiagnostic(method.CreateDiagnosticInfo(ConvertToLibraryImport, properties.ToImmutable(), method.Name).ToDiagnostic());
         }
 
+        private static bool IsBestFitMapping(IMethodSymbol method, DllImportData? dllImportData)
+        {
+            if (dllImportData.BestFitMapping == true)
+            {
+                return true;
+            }
+            AttributeData? bestFitMappingContainingType = method.ContainingType.GetAttributes().FirstOrDefault(attr => attr.AttributeClass.ToDisplayString() == TypeNames.System_Runtime_InteropServices_BestFitMappingAttribute);
+            if (bestFitMappingContainingType is not null)
+            {
+                return bestFitMappingContainingType.ConstructorArguments[0].Value is true;
+            }
+            AttributeData? bestFitMappingContainingAssembly = method.ContainingAssembly.GetAttributes().FirstOrDefault(attr => attr.AttributeClass.ToDisplayString() == TypeNames.System_Runtime_InteropServices_BestFitMappingAttribute);
```

Comment:
Would it be worth it to cache this in the StubEnvironment? Since it has a Compilation it'll never be incremental anyway, so we don't need to use a special dictionary (and we really could make it a class instead of a record at this point).

Reply:
I don't think so, because this is only used in the analyzer. If it were used in the generator, then putting it there would be useful.